pkg/operator/ceph/object/rgw.go

```diff
@@ -229,6 +288,22 @@ func (c *clusterConfig) startRGWPods(realmName, zoneGroupName, zoneName string,
 			if err != nil {
 				logger.Warningf("%v", err)
 			}
+			if c.store.Spec.AdminGateway.Enabled() {
+				depNameToRemove += "-admin"
+				if err := k8sutil.DeleteDeployment(c.clusterInfo.Context, c.context.Clientset, c.store.Namespace, depNameToRemove); err != nil {
+					logger.Warningf("error during deletion of deployment %q resource. %v", depNameToRemove, err)
+				}
+				secretToRemove := c.generateSecretName(k8sutil.IndexToName(depIDToRemove) + "-admin")
+				err = c.context.Clientset.CoreV1().Secrets(c.store.Namespace).Delete(c.clusterInfo.Context, secretToRemove, metav1.DeleteOptions{})
+				if err != nil && !kerrors.IsNotFound(err) {
+					logger.Warningf("failed to delete rgw secret %q. %v", secretToRemove, err)
+				}
+				err = c.deleteRgwCephObjects(depNameToRemove)
+				if err != nil {
+					logger.Warningf("%v", err)
+				}
+			}
+			i++
```

Comment:
If `i++` is at the end of the for loop, better to add it to the iterator on line 272 above. 

But I wonder why the increment was in the middle of the loop before? Definitely better not in the middle of the loop, unless I'm missing something.